python/ml4ir/applications/ranking/pipeline.py

```diff
@@ -124,6 +126,20 @@ def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
             logger=self.logger,
         )
 
+        # Adding REDUCE_LR_ON_PLATEAU as a callback
+        if 'lr_schedule' in self.model_config:
+            lr_schedule = self.model_config['lr_schedule']
+            lr_schedule_key = lr_schedule['key']
+            if lr_schedule_key == LearningRateScheduleKey.REDUCE_LR_ON_PLATEAU:
+                monitor_metric = self.args.monitor_metric
+                if not monitor_metric.startswith("val_"):
+                    monitor_metric = "val_{}".format(monitor_metric)
+                reduce_lr = tf.keras.callbacks.ReduceLROnPlateau(monitor=monitor_metric,
+                                                                 factor=lr_schedule.get('factor', 0.5),
+                                                                 patience=lr_schedule.get('patience', 1),
+                                                                 min_lr=lr_schedule.get('min_lr', 0.0001), verbose=1)
+                relevance_model.callbacks_list.append(reduce_lr)
+
```

Comment:
@mohazahran I had asked you to move this out of the pipeline. Add it to the model class that way we can use it with ml4ir as a library. pipelines are very restrictive.